samza-api/src/main/java/org/apache/samza/metadatastore/InMemoryMetadataStoreFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.metadatastore;
+
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.samza.config.Config;
+import org.apache.samza.metrics.MetricsRegistry;
+
+
+/**
+ * Factory for an in-memory {@link MetadataStore}. Not intended for use in production.
```

Comment:
samza-test has a bunch of other test utilities too that people depend on for local/integration testing (e.g. TestRunner). If users want to depend on this for testing, I think it would make sense for them to depend on samza-test directly.

Reply:
It's also used in `samza-core` unit tests.

Reply:
Thanks for clarifying that. I still feel that it is odd to put a class in samza-api which is only for testing. Is it a requirement for a metadata store to be persistent? Maybe we could just say that this is valid in production as long as you understand that it gets cleared on restart.